### ISSUE TITLE
Does not find matcher when missing automatically injected 'Accept' header

### ISSUE THREAD

**Issue by buddyp450** [ctimmerm/axios-mock-adapter#214]
Looking for design input here...

The current utils is an exact match which can lead to confusion when inserting headers on something like a `POST` (as I did) with `'Content-Type': 'application/json'` because my matcher wouldn't match that against the automatically injected `Accept` header which makes it `{ 'Accept': 'application/json, text/plain, */*', 'Content-Type': 'application/json' }`

https://github.com/ctimmerm/axios-mock-adapter/blob/master/src/utils.js#L45

**Comment by mkalam-alami** [comment]
We're having the same issue.

The injected `Content-Type` header brings an additional issue: for calls without parameters, it defaults to "application/x-www-form-urlencoded", while requests with bodies set "application/json;charset=utf-8".

This means currently we have two set of headers depending on the endpoint we test:

```typescript

const EXPECTED_HEADERS = {
  'Accept': 'application/json, text/plain, */*',
  'Authorization': `Bearer ${TOKEN}`,
  'Content-Type': 'application/json;charset=utf-8'
};
const EXPECTED_HEADERS_NO_BODY = {
  ...EXPECTED_HEADERS,
  'Content-Type': 'application/x-www-form-urlencoded'
};
```

PR #220 would be much welcome.

**Comment by ctimmerm** [comment]
Doing asymmetric matches has been merged in #181.
